feat(navigation): create provideDaffNavigationInMemoryDriver (#4031)
graycoreio/daffodil: libs/navigation/guides/driver/in-memory.md

# In Memory

> The `@daffodil/navigation` In-Memory driver provides randomly generated menus that your application can use as necessary to render trees of navigational elements. This driver is particularly useful for development, testing, and prototyping scenarios where you don't have a backend navigation service available.

Reviewer: ```suggestion
The `@daffodil/navigation` in-memory driver provides randomly generated menus for rendering navigational element trees. This driver is useful for development, testing, and prototyping when a backend navigation service is unavailable.
```

## Features

> - The in-memory driver uses Angular's `angular-in-memory-web-api` under the hood

Reviewer: ``` 
- Uses Angular's `angular-in-memory-web-api` under the hood
- Generates random data using `@daffodil/navigation/testing` factories
- Intended for development and testing only—use appropriate drivers for production environments
- Caches navigation trees in memory until page refresh
- Supports [custom seed data](/libs/navigation/guides/driver/in-memory.md#custom-seed-data) for persistent behavior
```

Author: 0eccfba941b61c724d534a787002c43c920ccd33

- Random data is generated using the `@daffodil/navigation/testing` factories
- The driver is intended for development and testing only - use appropriate production drivers for production environments
- Navigation trees are cached in memory and persist until refresh.
- [Custom Seed Data](#custom-seed-data) can be provided for more persistent behavior.

> ## Basic Usage

Reviewer: ```suggestion
## Basic usage
```

I have not been title casing headings unless it's necessary (e.g. Basic usage of Angular)

Author: 8d7cad2040b11946dedb95cfd1731b04600a1b0b

The recommended approach for modern Angular applications is to use the standalone provider function:

```ts
import { ApplicationConfig } from '@angular/core';
import { provideRouter } from '@angular/router';
import { provideHttpClient } from '@angular/common/http';
import { provideDaffInMemoryDriver } from '@daffodil/driver/in-memory';
import { provideDaffNavigationInMemoryDriver } from '@daffodil/navigation/driver/in-memory';

export const appConfig: ApplicationConfig = {
	providers: [
		provideRouter(routes),
		provideHttpClient(),
		provideDaffInMemoryDriver(myConfig),
		provideDaffNavigationInMemoryDriver(),
	],
};
```

> ## Custom Seed Data

Reviewer: ```suggestion
## Custom seed data
```

Author: 8d7cad2040b11946dedb95cfd1731b04600a1b0b

> By default, the in-memory driver generates random navigation data using the `DaffNavigationTreeFactory`. However, you can provide your own custom navigation tree data:

Reviewer: ```suggestion
By default, the in-memory driver generates random navigation data using `DaffNavigationTreeFactory`. You can also provide your own custom navigation tree data:
```

Author: 8d7cad2040b11946dedb95cfd1731b04600a1b0b

	id: 'root',
	url: '/',
	name: 'Main Navigation',
	breadcrumbs: [],
	children: [
		{
			id: 'products',
			url: '/products',
			name: 'Products',
			breadcrumbs: [{ id: 'root', name: 'Home', url: '/' }],
			children: [
				{
					id: 'category-1',
					url: '/products/category-1',
					name: 'Category 1',
					total_products: 25,
					breadcrumbs: [
						{ id: 'root', name: 'Home', url: '/' },
						{ id: 'products', name: 'Products', url: '/products' }
					],
					children: []
				},
				{
					id: 'category-2',
					url: '/products/category-2',
					name: 'Category 2',
					total_products: 15,
					breadcrumbs: [
						{ id: 'root', name: 'Home', url: '/' },
						{ id: 'products', name: 'Products', url: '/products' }
					],
					children: []
				}
			]
		},
		{
			id: 'about',
			url: '/about',
			name: 'About Us',
			breadcrumbs: [{ id: 'root', name: 'Home', url: '/' }],
			children: []
		}
	]
};

export const appConfig: ApplicationConfig = {
	providers: [
		provideRouter(routes),
		provideHttpClient(),
		provideDaffInMemoryDriver(myConfig),
		provideDaffNavigationInMemoryDriver(),
		provideDaffNavigationInMemorySeedDataProvider(() => customNavigationTree),
	],
};
```

> ## Common Use Cases

Reviewer: ```suggestion
## Common use cases
```

Author: 8d7cad2040b11946dedb95cfd1731b04600a1b0b

### Development Environment

> Use the in-memory driver during development to work without a backend:

Reviewer: ```suggestion
Use the in-memory driver to develop without a backend:
```

Author: 8d7cad2040b11946dedb95cfd1731b04600a1b0b
